docs/source/benchmark/submission.md

```diff
@@ -0,0 +1,97 @@
+
+We provide a base image at Dockerhub: `haosulab/mani-skill2:latest`. It is based on `nvidia/cudagl:11.3.1-devel-ubuntu20.04`. The corresponding Dockerfile is [here](https://github.com/haosulab/ManiSkill2/blob/main/docker/Dockerfile).
+
+Here is an example of how to customize the base docker image.
+
+```Dockerfile
+FROM haosulab/mani-skill2:latest
+
+# Install additional python packages you need
+RUN pip install pytorch==1.12.1
+
+# Copy your codes (including user_solution.py) and model weights
+COPY ${YOUR_CODES_AND_WEIGHTS_IN_HOST} ${YOUR_CODES_AND_WEIGHTS_IN_CONTAINER}
+ENV PYTHONPATH ${YOUR_CODES_AND_WEIGHTS_IN_CONTAINER}:$PYTHONPATH
+```
+
+Given a Dockerfile, you can build your submission docker image.
+
+```bash
+# It is suggested to run this command under the directory containing Dockerfile
+docker build -f ${PATH_TO_YOUR_DOCKERFILE} -t mani-skill2-submission
+```
+
+Finally, you can tag your image and push it to a public docker registry (e.g., [Dockerhub](https://hub.docker.com/)).
+
+```bash
+# Tag your local image before uploading it to Dockerhub
+docker tag mani-skill2-submission ${DOCKERHUB_USER_NAME}/mani-skill2-submission:${UNIQUE_TAG}
+docker push ${DOCKERHUB_USER_NAME}/mani-skill2-submission:${UNIQUE_TAG}
+```
+
+As Dockerhub's registry is public, we recommend you create a new anonymous docker account to prevent people from finding it.
+
+:::{warning}
+We only accept the docker image the size of which is smaller than 24GB. Please refer to <https://docs.docker.com/develop/develop-images/dockerfile_best-practices/> to reduce size.
+:::
+
+## Test the docker image locally
+
+Run the following script to test your docker image locally. If this works, it is ready for submission to the challenge.
+
+```bash
+export DOCKER_IMAGE=mani-skill2-submission:${UNIQUE_TAG}
+export CONTAINER_NAME=mani-skill2-evaluation
+# Initialize a detached container. If you are evaluating tasks with extra assets, you need to mount the directory containing downloaded assets to the container.
+docker run -d --rm --gpus all --name ${CONTAINER_NAME} \
+    -v ${PATH_TO_MS2_ASSET_DIR}:/data \
+    ${DOCKER_IMAGE}
+# Interactive debug
+docker exec -it ${CONTAINER_NAME} /bin/bash
+# Or run evaluation
+docker exec -it ${CONTAINER_NAME} /bin/bash -c "export MS2_ASSET_DIR=/data; python -m mani_skill2.evaluation.run_evaluation -e PickCube-v0 -o /eval_results/PickCube-v0 -n 1"
+# Finally, you can delete the container
+docker kill ${CONTAINER_NAME}
+```
+
+## Online Submission
+
+Once you have built and pushed a docker image, you are ready to submit to the competition. Go to the competition [submissions page](https://sapien.ucsd.edu/challenges/maniskill/challenges/ms2022/submit) and give your submission a name and enter the docker image name+tag (format: `registry.hub.docker.com/USERNAME/IMG_NAME:TAG`; Do not use the `latest` tag). Then select which track you are submitting to. Lastly, tick/untick which tasks you would like to evaluate your submission on.
```

Comment:
link should be https://sapien.ucsd.edu/challenges/maniskill/challenges/ms2/submit